state/upgrade.go

```diff
@@ -210,6 +211,42 @@ func (info *UpgradeInfo) isModelUUIDUpgradeDone() (bool, error) {
 	return n > 0, nil
 }
 
+// upgradeStatusHistoryAndOps sets the model's status history and returns ops for
+// setting model status according to the UpgradeStatus.
+func upgradeStatusHistoryAndOps(st *State, upgradeStatus UpgradeStatus, now int64) ([]txn.Op, error) {
+	model, err := st.Model()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	globalKey := model.globalKey()
```

Comment:
Don't do this. You are getting a database object for no reason. Just use the 'modelGlobalKey' constant.